avl/_admin.py

```diff
@@ -0,0 +1,377 @@
+        # Apply a restrictive policy to the "user manager" user. This policy
+        # restricts operations to user IDs under the AVL path, and forces the
+        # user manager to apply the permissions boundary to any users it
+        # creates, so it can't escalate its permissions by creating more
+        # powerful users.
+
+        user_pattern = f"{self.aws_account_id}:user/{self.resource_prefix}-" \
+                       f"{BUCKET_ACCESS_USER_PREFIX}/*"
+        self.iam_client.put_user_policy(
+            UserName=user_manager_username,
+            PolicyName="aws-user-manager-policy",
+            PolicyDocument=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Sid": "CreateUsersWithBoundaryAndPath",
+                            "Effect": "Allow",
+                            "Action": ["iam:CreateUser"],
+                            "Resource": user_pattern,
+                            "Condition": {
+                                "StringEquals": {
+                                    "iam:PermissionsBoundary":
+                                        f"{self.aws_account_id}:policy/"
+                                        f"{PERMISSIONS_BOUNDARY}"
+                                }
+                            },
+                        },
+                        {
+                            "Sid": "ManageUsersUnderPath",
+                            "Effect": "Allow",
+                            "Action": [
+                                "iam:GetUser",
+                                "iam:DeleteUser",
+                                "iam:PutUserPolicy",
+                                "iam:CreateAccessKey",
+                                "iam:DeleteAccessKey",
+                                "iam:ListAccessKeys"
+                            ],
+                            "Resource": user_pattern,
+                        },
+                        {
+                            "Sid": "ListUsers",
+                            "Effect": "Allow",
+                            "Action": ["iam:ListUsers"],
+                            "Resource": "*",
+                        },
+                    ],
+                }
+            ),
+        )
+
+        # Create and store credentials for the "user manager" user.
+        access_key = self.iam_client.create_access_key(
+            UserName=user_manager_username)
+        self.user_manager_key_id = access_key["AccessKey"]["AccessKeyId"]
+        self.user_manager_key_secret = access_key["AccessKey"][
+            "SecretAccessKey"]
+
+
+class BucketAccessUserCreator:
+    """Create temporary IAM users for AVL S3 bucket access
+
+    This class is intended to be used by the Jupyter Hub process when
+    spawning a new user environment. It creates (if not already present)
+    a temporary IAM user with permissions to read and write to the AVL
```

Comment:
```suggestion
    an  IAM user with permissions to read and write to the AVL
```
If I remember correctly, we decided, that the IAM users are not temporary, but created once and then kept?

Reply:
I think we left it open -- inactive IAM users can in principle be culled at any time without any noticeable effect, because they are recreated whenever needed, but at present the culling isn't yet implemented because there aren't enough users to make it necessary. But you're right, the term "temporary" is probably confusing in this context because IAM users aren't temporary in the same sense as roles, which have a predefined, limited lifespan. I'll update the documentation to make this clearer.

Reply:
I've replaced "temporary user" with "bucket access user" throughout, and added more clarification about how the IAM users are created and (possibly) deleted.